### PR TITLE
feat(frontend): [PYTA-16] add register page

### PR DESCRIPTION
**Pull request by mateusz-kurowskiUG**
<sub><a href="https://huly.app/guest/pyta?token=eyJ0eXAiOiJKV1QiLCJhbGciOiJIUzI1NiJ9.eyJsaW5rSWQiOiI2NzY0NzU5NGYxNDRmNTg1YWU0ZTRmZTYiLCJndWVzdCI6InRydWUiLCJlbWFpbCI6IiNndWVzdEBoYy5lbmdpbmVlcmluZyIsIndvcmtzcGFjZSI6InctbWVtZXRlbHZlLXB5dGEtNjcwYzBkMWItY2Y5ZTE4MDAyYS00NDk4ZGIifQ.x-CKJ5fBbPvykTOGEiSFeye6cRMGYpv56kxwMOAdfhc">Huly&reg;: <b>PYTA-79</b></a></sub>

### CODE REVIEW

File: frontend/src/hooks/use-toast.ts [hunk 1]
```diff
@@ -0,0 +1,191 @@
+"use client";
+
+// Inspired by react-hot-toast library
```
Comment (Memetelve) [inline frontend/src/hooks/use-toast.ts:3]: ??

File: frontend/src/app/[locale]/auth/page.tsx [hunk 2]
```diff
@@ -1,13 +1,19 @@
...
-		</div>
-	);
+	const router = useRouter();
+	const path = usePathname();
+
+	useEffect(() => {
+		if (path === "/auth") {
+			router.push("/auth/sign-in");
+		}
+	}, [path, router]); // Dependencies to ensure useEffect runs on path change
```
Comment (Memetelve) [inline frontend/src/app/[locale]/auth/page.tsx:14]: redundant comment

File: frontend/src/app/[locale]/auth/layout.tsx [hunk 3]
```diff
@@ -0,0 +1,47 @@
...
+	const locale = useLocale();
+
+	const activeTab =
+		pathname === `/${locale}/auth/sign-up` ? "signUp" : "signIn";
+
+	return (
+		<div className="flex items-center justify-center h-svh">
+			<div className="w-[400px] mx-2">
+				<div className="flex border-b border-gray-200 mb-4">
+					{/* Links replacing Tabs */}
```
Comment (Memetelve) [inline frontend/src/app/[locale]/auth/layout.tsx:19]: ??

File: frontend/src/app/[locale]/auth/_components/DatePicker.tsx [hunk 4]
```diff
@@ -0,0 +1,165 @@
...
+export function DatePicker({ fromDate, toDate }: DatePickerProps) {
+	const [date, setDate] = React.useState<Date>(toDate);
+
+	const t = useTranslations("SignUpForm.Calendar");
+
+	const locale = useLocale();
+
+	const calendarLocale = locale === "pl" ? pl : enUS;
+
+	const months = [
```
Comment (Memetelve) [inline frontend/src/app/[locale]/auth/_components/DatePicker.tsx:48]: is there a way to maybe not do that? create SignUpForm.Calendar.Months and iterate through all elements?

File: frontend/src/app/[locale]/auth/_components/DatePicker.tsx [hunk 5]
```diff
@@ -0,0 +1,165 @@
...
+}
+
+export function DatePicker({ fromDate, toDate }: DatePickerProps) {
+	const [date, setDate] = React.useState<Date>(toDate);
+
+	const t = useTranslations("SignUpForm.Calendar");
+
+	const locale = useLocale();
+
+	const calendarLocale = locale === "pl" ? pl : enUS;
```
Comment (Memetelve) [inline frontend/src/app/[locale]/auth/_components/DatePicker.tsx:46]: leaves us no room for new languages

File: frontend/src/app/[locale]/auth/_components/DatePicker.tsx [hunk 6]
```diff
@@ -0,0 +1,165 @@
...
+	SelectValue,
+} from "@/components/ui/select";
+import { cn } from "@/lib/utils";
+import { format, getMonth, getYear, setMonth, setYear } from "date-fns";
+import { Calendar as CalendarIcon } from "lucide-react";
+import { useLocale, useTranslations } from "next-intl";
+import * as React from "react";
+import { useFormContext } from "react-hook-form";
+
+// Import the necessary locales from date-fns
```
Comment (Memetelve) [inline frontend/src/app/[locale]/auth/_components/DatePicker.tsx:31]: ??

File: frontend/src/app/[locale]/auth/_components/DatePicker.tsx [hunk 7]
```diff
@@ -0,0 +1,165 @@
...
+								toDate={toDate}
+								selected={date}
+								onSelect={(selectedDate) => {
+									field.onChange(selectedDate);
+									handleSelect(selectedDate);
+								}}
+								initialFocus
+								month={date}
+								onMonthChange={setDate}
+								locale={calendarLocale} // Pass the locale to the Calendar component
```
Comment (Memetelve) [inline frontend/src/app/[locale]/auth/_components/DatePicker.tsx:156]: yeah, obvious

File: frontend/src/app/[locale]/auth/_components/SignUpForm.tsx [hunk 8]
```diff
@@ -14,93 +11,238 @@ import {
...
+import { useRouter } from "@/i18n/routing";
+import { useSignupUser } from "@/query/auth";
+import { zodResolver } from "@hookform/resolvers/zod";
+import type { AxiosError } from "axios";
+import moment from "moment";
 import { useTranslations } from "next-intl";
+import { FormProvider, useForm } from "react-hook-form";
+import { toast } from "sonner";
+import { z } from "zod";
+import { passwordRegex } from "../../../../lib/regex";
```
Comment (Memetelve) [inline frontend/src/app/[locale]/auth/_components/SignUpForm.tsx:23]: maybe use @?

File: frontend/src/app/[locale]/auth/_components/SignUpForm.tsx [hunk 9]
```diff
@@ -14,93 +11,238 @@ import {
...
+import { DatePicker } from "./DatePicker";
 
 const SignUpForm = () => {
 	const t = useTranslations("SignUpForm");
-	// TODO: Add password regex
-	const formSchema = z
+	const router = useRouter();
+	const now = new Date();
+	const minBirthDate = new Date(now);
```
Comment (Memetelve) [inline frontend/src/app/[locale]/auth/_components/SignUpForm.tsx:30]: why pass now as a param?

File: frontend/src/app/[locale]/auth/_components/SignUpForm.tsx [hunk 10]
```diff
@@ -14,93 +11,238 @@ import {
...
+	const minAge = 18;
+	const maxAge = 140;
+
+	// Calculate the minBirthDate (18 years ago)
+	minBirthDate.setFullYear(now.getFullYear() - minAge);
+
+	// Calculate the maxBirthDate (140 years ago)
+	const maxBirthDate = new Date(now);
```
Comment (Memetelve) [inline frontend/src/app/[locale]/auth/_components/SignUpForm.tsx:38]: move this next to now and minDate

File: frontend/src/app/[locale]/auth/layout.tsx [hunk 11]
```diff
@@ -0,0 +1,47 @@
...
+import { usePathname } from "next/navigation";
+import type React from "react";
+
+const Layout = ({ children }: { children: React.ReactNode }) => {
+	const t = useTranslations("AuthPage");
+	const pathname = usePathname();
+	const locale = useLocale();
+
+	const activeTab =
+		pathname === `/${locale}/auth/sign-up` ? "signUp" : "signIn";
```
Comment (Memetelve) [inline frontend/src/app/[locale]/auth/layout.tsx:13]: use endsWith instead of creating a formatted string

**Comment by Memetelve** [comment]
can we start using some form of import sorter?

File: frontend/messages/en.json [hunk 12]
```diff
@@ -7,17 +7,65 @@
...
+		"registerButtonText": "Register",
+
+		"messages": {
+			"required": "{field} is required.",
+			"tooLong": "{field} is too long.",
+			"notAnEmail": "Surely does not look like an email.",
+			"tooYoung": "You are too young to use the application.",
+			"tooOld": "You are probably too old to be alive, is that date correct?",
+			"noPasswordMatchError": "The passwords did not match.",
+			"acceptTerms": "Accepting terms",
```
[the remark this answers is not in the document]
Comment (mateusz-kurowskiUG) [reply frontend/messages/en.json:19]: you are improper

File: frontend/src/app/[locale]/auth/_components/SignUpForm.tsx (hunk 10 above)
Comment (mateusz-kurowskiUG) [reply frontend/src/app/[locale]/auth/_components/SignUpForm.tsx:38]: No.

File: frontend/src/app/[locale]/auth/_components/SignUpForm.tsx (hunk 9 above)
Comment (mateusz-kurowskiUG) [reply frontend/src/app/[locale]/auth/_components/SignUpForm.tsx:30]: image

File: frontend/src/app/[locale]/auth/_components/DatePicker.tsx (hunk 5 above)
Comment (mateusz-kurowskiUG) [reply frontend/src/app/[locale]/auth/_components/DatePicker.tsx:46]: ok

File: frontend/src/app/[locale]/auth/_components/DatePicker.tsx (hunk 4 above)
Comment (mateusz-kurowskiUG) [reply frontend/src/app/[locale]/auth/_components/DatePicker.tsx:48]: no.

File: frontend/messages/en.json [hunk 13]
```diff
@@ -7,17 +7,65 @@
...
-		"passwordTooLongError": "Password too long.",
-		"noPasswordMatchError": "The passwords did not match.",
-		"registerButtonText": "Sign Up",
-		"emailInputLabel": "Your Email",
-		"passwordInputLabel": "Password",
-		"confirmPasswordInputLabel": "Confirm Password",
-		"emailPlaceHolder": "Email",
-		"passwordPlaceHolder": "Password",
-		"confirmPasswordPlaceHolder": "Password"
+		"registerButtonText": "Register",
```
[the remark this answers is not in the document]
Comment (mateusz-kurowskiUG) [reply frontend/messages/en.json:10]: rename what

[the remark this answers is not in the document]
Comment (Memetelve) [reply frontend/messages/en.json:10]: the json key

[the remark this answers is not in the document]
Comment (mateusz-kurowskiUG) [reply frontend/messages/en.json:10]: to what?

**Review by olszewskista — APPROVED** [review]

**Review by Memetelve — APPROVED** [review]

**Pull request merged**